src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionItemResolver.cs

```diff
@@ -39,7 +39,30 @@ internal class RazorCompletionItemResolver : CompletionItemResolver
             return null;
         }
 
-        var associatedRazorCompletion = razorCompletionResolveContext.CompletionItems.FirstOrDefault(completion => string.Equals(completion.DisplayText, completionItem.Label, StringComparison.Ordinal));
+        var associatedRazorCompletion = razorCompletionResolveContext.CompletionItems.FirstOrDefault(completion =>
+        {
+            if (!string.Equals(completion.DisplayText, completionItem.Label, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            // We may have items of different types with the same label (e.g. snippet and keyword)
+            if (clientCapabilities is not null)
+            {
+                // CompletionItem.Kind and RazorCompletionItem.Kind are not compatible/comparable, so we need to convert
+                // Razor completion item to VS completion item (as logic to convert just the kind is not easy to separate from
+                // the rest of the conversion logic) prior to comparing them
+                RazorCompletionListProvider.TryConvert(completion, clientCapabilities, out var convertedRazorCompletionItem);
+                if (convertedRazorCompletionItem != null)
```

Comment:
The `TryConvert` method is intended to be used like so.

```suggestion
                if (RazorCompletionListProvider.TryConvert(completion, clientCapabilities, out var convertedRazorCompletionItem))
```

Reply:
Most definitely. Not sure what I was thinking there. :) Thanks!